### PR TITLE
Let users download profiles as vCards

### PR DESCRIPTION
**Pull request by EvanHahn**
Users can now download a simple [vCard] for profiles they visit. This is implemented as a new route.

App screenshot

Here's how it looks imported into [GNOME Contacts]:

GNOME Contacts

And into iOS:

<p align="center">iOS screenshot</p>

It uses [VObject], the best Python vCard package I could find. Some work was required to make edge cases work, such as the legacy vCard "name" field.

Addresses #106.

[vCard]: https://en.wikipedia.org/wiki/VCard
[VObject]: https://eventable.github.io/vobject/
[GNOME Contacts]: https://apps.gnome.org/Contacts/

### CODE REVIEW

File: cl8/users/models.py [hunk 1]
```diff
@@ -284,6 +286,23 @@ def generate_invite_mail(self):
 
         return {"text": rendered_invite_txt, "html": rendered_invite_html}
 
+    def vcard(self) -> vobject.vCard:
+        properties = {
```
Comment (EvanHahn) [inline cl8/users/models.py:290]: We could add additional properties in the future, but I think this is a good starter set. The [vCard Wikipedia page](https://en.wikipedia.org/wiki/VCard#Properties) has a good list of these properties, as does [the spec](https://datatracker.ietf.org/doc/html/rfc6350).

File: cl8/utils/tests/test_vcard_util.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class TestVcardUtil:
+    def test_full_name_to_name(self):
+        all_in_given_name = [
+            "",
+            "Biscuit",
+            "Dr.",
+            "Dr. Gravy Biscuit",
+            "Sweet Gravy Biscuit",
```
Comment (mrchrisadams) [inline cl8/utils/tests/test_vcard_util.py:14]: Good names.

File: cl8/utils/tests/test_vcard_util.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import pytest
+from vobject.vcard import Name
+
+from .. import vcard_util
+
+
+class TestVcardUtil:
+    def test_full_name_to_name(self):
```
Comment (mrchrisadams) [inline cl8/utils/tests/test_vcard_util.py:8]: Would you mind adding a doc string giving a very short summary of what behaviour the test is checking is in place? Something like:

> check that we can free-form full name strings to structured Vcard names

or along those lines - assuming this is exercising `full_name_to_name()`

File: cl8/utils/tests/test_vcard_util.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            ("Gravy", "Biscuit"),
+            ("Großer", "Keks"),
+        ]
+        for given, family in split:
+            full_name = f"{given} {family}"
+            actual = vcard_util.full_name_to_name(full_name)
+            expected = Name(given=given, family=family)
+            assert actual == expected
+
+    def test_content_disposition_filename(self):
```
Comment (mrchrisadams) [inline cl8/utils/tests/test_vcard_util.py:35]: Same here. I know the `content_disposition_filename()` is quite well documented, but a one liner along the lines of:

> check that we can create a valid filename for a vcard, based on a profile

Or something to that effect.

**Review by mrchrisadams — APPROVED** [review]
Thanks for this @EvanHahn , this looks great! I'm very happy to merge this in. - I've asked for a couple of docstrings for the tests as a convenience but otherwise I'm very happy for this to be merged in.

I'm not sure you gave the ability to merge in from an approved PR, if not please let me know, and I'll either grant access or merge it in at your end.

File: cl8/utils/tests/test_vcard_util.py (hunk 4 above)
Comment (EvanHahn) [reply cl8/utils/tests/test_vcard_util.py:39]: Done in 398261fabd55a5cb670fb9c88dd1ad6a43e20a06.

File: cl8/utils/tests/test_vcard_util.py (hunk 3 above)
Comment (EvanHahn) [reply cl8/utils/tests/test_vcard_util.py:8]: Done in 398261fabd55a5cb670fb9c88dd1ad6a43e20a06.

**Comment by EvanHahn** [comment]
I think I've addressed all comments!

I don't have permission to merge this myself. Happy to have you do it, or to get the permission to do so myself—whichever is easier for you.

**Review by mrchrisadams — COMMENTED** [review]
Very happy! Thanks @EvanHahn

**Pull request merged**